osu.Game/Overlays/BeatmapListing/BeatmapListingSearchControl.cs

```diff
@@ -136,10 +141,19 @@ public BeatmapListingSearchControl()
             categoryFilter.Current.Value = SearchCategory.Leaderboard;
         }
 
+        private IBindable<bool> allowExplicitContent;
+
         [BackgroundDependencyLoader]
-        private void load(OverlayColourProvider colourProvider)
+        private void load(OverlayColourProvider colourProvider, OsuConfigManager config)
         {
             background.Colour = colourProvider.Dark6;
+
+            allowExplicitContent = config.GetBindable<bool>(OsuSetting.AllowExplicitContent);
+            allowExplicitContent.BindValueChanged(allow =>
+            {
+                // Update search control if global "explicit allowed" setting changed.
```

Comment:
sort of a redundant comment

Reply:
thought it's kinda unclear given that the tab control can actually be changed by user input, sure about removing it?

Reply:
i don't see anything unclear in this code? there's a setting, and stuff happens when that setting changes.

Reply:
hmm, alright i'll take that into consideration in future PRs (in-line comments are the only things that I always don't know whether to put or not)